dask/array/utils.py

```diff
@@ -565,6 +565,22 @@ def svd_flip(u, v, u_based_decision=False):
     return u, v
 
 
+################
+# scipy.linalg #
+################
+
+
+def solve_triangular_safe(a, b, lower=False):
+    import scipy.linalg
+
+    if is_cupy_type(a):
+        import cupyx.scipy.linalg
+
+        return cupyx.scipy.linalg.solve_triangular(a, b, lower=lower)
+
+    return scipy.linalg.solve_triangular(a, b, lower=lower)
```

Comment:
There are other `scipy.linalg` functions that could use the same idea. Could we generalize this?

I haven't checked them, but if the first argument is always an ndarray, we could do something like the following:

```python
def scipy_linalg_safe(func_name, *args, **kwargs):
    if is_cupy_type(a):
        import cupyx.scipy.linalg
        func = getattr(cupyx.scipy.linalg, func_name)
    else:
        import scipy.linalg
        func = getattr(scipy.linalg, func_name)

    return func(*args, **kwargs)
```

And the caller would then obviously call `scipy_linalg_safe("solve_triangular", a, b, lower=lower)`.

Reply:
I think this is a good idea generally but at the moment dask, I think, only uses three functions from scipy.linalg: `solve_triangluar`, `lu`, `cholesky`.  `solve_traingluar` is currently supported in `cupyx.scipy.linalg` and `cholesky` is from cupy.linalg.  In my last commit I added support for `cholesky`

Reply:
Might still be worth doing this for `lu` later, as both are part of `cupyx.scipy.linalg`.

Reply:
Ah, I was looking at CuPy 8.  `lu` is in CuPy 9 so I'll add this in as you suggested